docker/ray-ml/Dockerfile

```diff
@@ -1,22 +1,28 @@
 ARG BASE_IMAGE
-FROM rayproject/ray:nightly"$BASE_IMAGE"
+ARG FROM=rayproject/ray:nightly"$BASE_IMAGE"
+FROM "$FROM"
+
+ARG REQUIREMENT_PATH=.
+ARG DOCKER_REQUIREMENT_PATH=.
+ARG ML_REQUIREMENT_PATH=.
+ARG SCRIPT_PATH=.
 
 # We have to uninstall wrapt this way for Tensorflow compatibility
-COPY requirements.txt ./
-COPY requirements_compiled.txt ./
-COPY dl-cpu-requirements.txt ./
-COPY dl-gpu-requirements.txt ./
-COPY ray-docker-requirements.txt ./
-COPY core-requirements.txt ./
-COPY data-requirements.txt ./
-COPY rllib-requirements.txt ./
-COPY rllib-test-requirements.txt ./
-COPY tune-requirements.txt ./
-COPY tune-test-requirements.txt ./
-COPY train-requirements.txt ./
-COPY train-test-requirements.txt ./
+COPY ${REQUIREMENT_PATH}/requirements.txt ./
+COPY ${REQUIREMENT_PATH}/requirements_compiled.txt ./
+COPY ${ML_REQUIREMENT_PATH}/dl-cpu-requirements.txt ./
+COPY ${ML_REQUIREMENT_PATH}/dl-gpu-requirements.txt ./
+COPY ${DOCKER_REQUIREMENT_PATH}/ray-docker-requirements.txt ./
+COPY ${ML_REQUIREMENT_PATH}/core-requirements.txt ./
+COPY ${ML_REQUIREMENT_PATH}/data-requirements.txt ./
+COPY ${ML_REQUIREMENT_PATH}/rllib-requirements.txt ./
+COPY ${ML_REQUIREMENT_PATH}/rllib-test-requirements.txt ./
+COPY ${ML_REQUIREMENT_PATH}/tune-requirements.txt ./
+COPY ${ML_REQUIREMENT_PATH}/tune-test-requirements.txt ./
+COPY ${ML_REQUIREMENT_PATH}/train-requirements.txt ./
+COPY ${ML_REQUIREMENT_PATH}/train-test-requirements.txt ./
 
-COPY install-ml-docker-requirements.sh ./
+COPY ${SCRIPT_PATH}/install-ml-docker-requirements.sh ./
```

Comment:
this will break the old docker build path, right? which will build the other containers still?

maybe do a rename on that first?

Reply:
ah.. ok. maybe it is just easier to change the build file.. feels that now it has a lot of args that are not really necessary, but just for backwards compatibility purposes..

Reply:
the existing logic of build-docker-images.py is to copy all requirements file into one place and call docker build in that root (https://github.com/ray-project/ray/blob/master/ci/build/build-docker-images.py#L494 and https://github.com/ray-project/ray/blob/master/ci/build/build-docker-images.py#L268); while wanda uses ray check out as root

the other way is to fix build-docker-images.py to use ray as root which is actually more complex to do

we can deprecate the whole build-docker-images.py thing after the migration to wanda is completed; once that is done, this Docker file can be simplified back to support wanda only